src/test/java/com/cottonfactory/products/controllers/TieControllerUnitTest.java

```diff
@@ -0,0 +1,60 @@
+package com.cottonfactory.products.controllers;
+
+
+import com.cottonfactory.products.entities.Tie;
+import com.cottonfactory.products.services.TieService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@WebMvcTest(TieController.class)
+public class TieControllerUnitTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @MockBean
+    TieService mockService;
+
+    @Test
+    public void test_createTie() throws Exception {
+        Tie tie = new Tie();
+        tie.setId(1);
+        tie.setColor("blue");
+        tie.setMaterial("silk");
+        tie.setType("bow tie");
+        when(mockService.createTie(any())).thenReturn(tie);
+
+        Tie tie2 = new Tie();
```

Comment:
Not clear why you have 2 ties in this test. It appears to be more than you need. Can you explain?

Reply:
tie object has id set in it and that will returned by the mock service. tie2 object does not have id in it. So when the tie2 is sent to controller, we are asserting whether the object(tie, first object) returned has id in it